packages/perseus/src/widgets/interactive-graphs/reducer/interactive-graph-reducer.ts

```diff
@@ -197,6 +206,40 @@ function doMovePoint(
     }
 }
 
+function doChangeSnapStep(
+    state: InteractiveGraphState,
+    action: ChangeSnapStep,
+): InteractiveGraphState {
+    if (
+        // Deep equality check since these are arrays
+        _.isEqual(state.snapStep, action.snapStep)
+    ) {
+        return state;
+    }
```

Comment:
I like this — it's an optimization to prevent unnecessary re-renders, right? It makes me wonder if we should do an `isEqual` check on every action, perhaps right before returning from the root reducer function:

```ts
if (_.isEqual(newState, oldState)) {
  return oldState;
}

return newState;
```